File: TPs/TP1/TP01Q02 - Palíndromo em C/palindromo.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool isFim(char s[]){
    return (strlen(s) == 3 && s[0] == 'F' && s[1] == 'I' && s[2] == 'M');
}

bool isPalindromo(char s[]){
    bool resp = true;
    int j = 0; 
    for(int i = strlen(s) - 1; i >=0 && resp; i--, j++){
        if(s[i] != s[j])
            resp = false;
    }
    return resp;
}

int main(){
    char entrada[1000][100];
    int numEntrada = 0;

    do{
        scanf(" %[^\n]s", entrada[numEntrada]);
    } while(isFim(entrada[numEntrada++]) == false);
    numEntrada--;

    for(int i = 0; i < numEntrada; i++){
        if(isPalindromo(entrada[i]) == true)
            printf("SIM\n");
        else
            printf("NAO\n");
    }
    return 0;
}

File: TPs/TP1/TP01Q01 - Palíndromo em Java/Palindromo.java
class Palindromo{ 
    public static boolean isFim(String s){
        return (s.length() == 3 && s.charAt(0) == 'F' && s.charAt(1) == 'I' && s.charAt(2) == 'M');
    }

    public static boolean isPalindromo(String s){
        boolean resp = true;
        int j = 0; 
        for(int i = s.length() - 1; i >= 0 && resp; i--, j++){
            if(s.charAt(i) != s.charAt(j))
                resp = false;
        }
        return resp;
    }

    public static void main (String[] args){
        String[] entrada = new String[1000];
        int numEntrada = 0;

        //Leitura da entrada padrao
        do {
            entrada[numEntrada] = MyIO.readLine();
        } while (isFim(entrada[numEntrada++]) == false);
        numEntrada--;   //Desconsiderar ultima linha contendo a palavra FIM
        
        for(int i = 0; i < numEntrada; i++){
            if(isPalindromo(entrada[i]) == true)
                MyIO.println("SIM");
            else
                MyIO.println("NAO");
        }
    }
}